gammapy/makers/utils.py

```diff
@@ -35,23 +35,43 @@ def make_map_exposure_true_energy(pointing, livetime, aeff, geom):
         Effective area
     geom : `~gammapy.maps.WcsGeom`
         Map geometry (must have an energy axis)
+    average_over_region: Bool
+        If geom is a RegionGeom, whether to just
+        consider the values at the region center
+        or the average over the whole region
 
     Returns
     -------
     map : `~gammapy.maps.WcsNDMap`
         Exposure map
     """
-    offset = geom.separation(pointing)
-    energy_true = geom.axes["energy_true"].center
+    if average_over_region:
+        wcs_geom = geom.to_wcs_geom().upsample(2)
```

Comment:
Why the factor of 2 here? Is there a criterion to define it a priori?

Reply:
I agree here I think the oversampling by a factor of 2 is an arbitrary choice, so one could either make it an argument (possibly `region_oversampling`, like `background_oversampling`), or what I already proposed to @LauraOlivera, was to use a set of weights to represent the region, like:

<img width="395" alt="image" src="https://user-images.githubusercontent.com/3715928/106108047-09823580-6148-11eb-8170-c7940a09f282.png">

This set of weights can be pre-computed an cached on the `RegionGeom` and should help to improve the the accuracy of the integration a lot...

Reply:
Completely agree. I just had not decided yet if I wanted to implement this here on in a separate PR. First I wanted to make sure I was doing things right without weights, and I plan to introduce it next

Reply:
OK indeed. 
I suppose one question is the default `binsz` that is defined on the `RegionGeom`. Is there a way to determine it a priori? e.g. `radius` or `width` /10?

Reply:
Right now it's hard coded to `binsz=0.01` (https://github.com/gammapy/gammapy/blob/master/gammapy/maps/region.py#L48), this isn't ideal but in practice a choice that works for many cases I guess. We could add a parameter `RegionGeom(..., binsz_wcs=)`, which allows advanced users to modify the bin size to either improve accuracy or performance for large regions. By default we just work with `0.02` or `0.01` and the oversampling weights, I proposed below.

Reply:
I'm not sure we can come up with a generic solution like `width/10`, because regions can have all sorts of shapes...

Reply:
I agree that there should be an option to specify the bin size of the underlying `WcsGeom`. I tried to think of ways to get that info from the region size but I ran into the problem Axel describes below, because it might be tricky with weird region shapes. I think a reasonable default (0.01) + the weights is the way to go.

Reply:
The only way I would see is to add some kind of system, where we declare for each shape, which bin size to use:

```python3
def get_region_binsz(region):
    if isinstance(region, CircleSkyRegion):
        return 0.1 * region.radius
    elif isinstance(region , ...):
        return 0.2 * region.width
    elif isinstance(region, CompoundRegion):
        regions = compound_region_to_list(region)
        return np.min([get_region_binsz(region) for region in regions])
    else:
        raise ValueError(f"Region {region} not supported")
```
I'm not sure whether it's worth it...